tests/unit/app/models/test_config.py

```diff
@@ -312,9 +312,175 @@ def test_provider_config():
     assert "deployment_name is required" in str(excinfo.value)
 
 
+def test_that_url_is_required_provider_parameter():
+    """Test that provider-specific URL is required attribute."""
+    # provider type is set to "azure_openai"
+    with pytest.raises(ValidationError, match="url"):
+        ProviderConfig(
+            {
+                "name": "test_name",
+                "type": "azure_openai",
+                "url": "test_url",
+                "credentials_path": "tests/config/secret.txt",
+                "deployment_name": "deploment-name",
+                "azure_openai_config": {
+                    "tenant_id": "tenant-ID",
+                    "client_id": "client-ID",
+                    "client_secret_path": "tests/config/secret.txt",
+                    "credentials_path": "tests/config/secret.txt",
+                    "deployment_name": "deployment-name",
+                },
+                "models": [
+                    {
+                        "name": "test_model_name",
+                        "url": "test_model_url",
+                        "credentials_path": "tests/config/secret.txt",
+                    }
+                ],
+            }
+        )
+
+    # provider type is set to "openai"
+    with pytest.raises(ValidationError, match="url"):
+        ProviderConfig(
+            {
+                "name": "test_name",
+                "type": "openai",
+                "url": "test_url",
+                "credentials_path": "tests/config/secret.txt",
+                "project_id": "test_project_id",
+                "openai_config": {
+                    "credentials_path": "tests/config/secret.txt",
+                },
+                "models": [
+                    {
+                        "name": "test_model_name",
+                        "url": "test_model_url",
+                        "credentials_path": "tests/config/secret.txt",
+                    }
+                ],
+            }
+        )
+
+    # provider type is set to "bam"
+    with pytest.raises(ValidationError, match="url"):
+        ProviderConfig(
+            {
+                "name": "test_name",
+                "type": "bam",
+                "url": "test_url",
+                "credentials_path": "tests/config/secret.txt",
+                "project_id": "test_project_id",
+                "bam_config": {
+                    "credentials_path": "tests/config/secret.txt",
```

Comment:
I'm confused a bit what should be the right way to provide `credentials_path` then? Should it be on the provider root or <provider>_config?

Reply:
in future, the provider config will be correct way. ATM we are going to support "old" style as well, because it needs to be coordinated with OLS operator?

Reply:
I see, so is there some todo with jira about "disabling" the old way?

Reply:
added @onmete